lib/install/validate/network.go

```diff
@@ -169,6 +172,23 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *config.
 	var e *executor.NetworkEndpoint
 	var err error
 
+	// client and management networks need to have at least one
+	// routing destination if gateway was specified
+	for nn, n := range map[string]*data.NetworkConfig{
+		"client":     &input.ClientNetwork,
+		"management": &input.ManagementNetwork,
+	} {
+		if !ip.IsUnspecifiedIP(n.Gateway.IP) && len(n.Destinations) == 0 {
+			v.NoteIssue(fmt.Errorf("%s network gateway specified without at least one routing destination", nn))
+		}
+	}
+
+	// external network should not have any routing destinations specified
+	// if a gateway was specified
+	if !ip.IsUnspecifiedIP(input.ExternalNetwork.Gateway.IP) && len(input.ExternalNetwork.Destinations) > 0 {
+		v.NoteIssue(errors.New("external network must not have any routing destinations specified for gateway"))
```

Comment:
Maybe change the error here to "external network has the default route and must not have ...